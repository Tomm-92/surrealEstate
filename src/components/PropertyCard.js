import React from "react";
import "../styles/property-card.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

const PropertyCard = (props) => {
  const { title, type, bathrooms, bedrooms, price, city, email } = props;
  return (
    <div className="property-card">
      <FontAwesomeIcon icon="house" />
      <div className="property-title">{title}</div>
      <div className="property-city">
        {type} - {city}
      </div>
      <div className="property-bathrooms">{bathrooms}</div>
      <FontAwesomeIcon icon="bath" />
      <div className="property-bedrooms">{bedrooms}</div>
      <FontAwesomeIcon icon="bed" />
      <div className="property-price">{price}</div>
      <div className="property-email">
        <FontAwesomeIcon icon="envelope" />
        <p>
          <a href={`mailto:${email}`}>Send email</a>
        </p>
      </div>
    </div>
  );
};

export default PropertyCard;
