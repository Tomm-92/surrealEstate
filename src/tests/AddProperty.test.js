import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("AddProperty", () => {
  const validProps = {
    title: "Stub Title",
    type: "Stub Type",
    bathrooms: 2,
    bedrooms: 3,
    price: 100000,
    city: "Manchester",
    email: "M.Salah@gmail.com",
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for the props", () => {
    const { getByText } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(getByText("Stub Title")).toHaveClass("property-title");
    expect(getByText("Stub Type")).toHaveClass("property-type");
    expect(getByText("2")).toHaveClass("property-bathrooms");
    expect(getByText("3")).toHaveClass("property-bedrooms");
    expect(getByText("100000")).toHaveClass("property-price");
    expect(getByText("M.Salah@gmail.com")).toHaveClass("property-email");
  });
});
