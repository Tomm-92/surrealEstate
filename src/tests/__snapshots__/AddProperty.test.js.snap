// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-title"
  >
    Stub Title
  </div>
  <div
    class="property-type"
  >
    Stub Type
  </div>
  <div
    class="property-bathrooms"
  >
    2
  </div>
  <div
    class="property-bedrooms"
  >
    3
  </div>
  <div
    class="property-price"
  >
    100000
  </div>
  <div
    class="property-city"
  >
    Manchester
  </div>
  <div
    class="property-email"
  >
    M.Salah@gmail.com
  </div>
</DocumentFragment>
`;
